### PR TITLE
Fix: gather_all_tensors cross GPUs  in DDP

### LINKED ISSUE #3253: **gather_all_tensors_if_available**  share the same underlying storage for all GPUs
## 🐛 Bug

Hi, one of new features in   #2528  `gather_all_tensors_if_available` has a  `list` copy bug,  and this would lead that tensors in all GPUs  are the wrongly same as one GPU,  since they share the same storage:

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/converters.py#L304

```
gathered_result = world_size * [torch.zeros_like(result)]
```
change into:
```
gathered_result = [torch.zeros_like(result) for _ in range(world_size)]
```

### PR DESCRIPTION
**Pull request by ShomyLiu**
## What does this PR do?

This PR fixed the sharing the same underlying storage for all GPUs in `gather_all_tensors` function.

Fixes #3253 

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by s-rog — APPROVED** [review]
LGTM

File: pytorch_lightning/metrics/converters.py
Comment (SkafteNicki) [inline pytorch_lightning/metrics/converters.py:304]: Could you add a test to `tests/metrics/test_converters.py` that actually test that this function does what it is expected to do?

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/metrics/converters.py
Comment (ShomyLiu) [reply pytorch_lightning/metrics/converters.py:304]: Yeah, thanks for the advice, I will add a test for this RP.

**Review by Borda — CHANGES_REQUESTED** [review]
waiting for a test to be added :]

**Comment by ShomyLiu** [comment]
@SkafteNicki   Hi, I found that the commit history has been disordered when I am going to update a test case in my PR. 
I'd like to close this PR and reopen a new PR for a clear commit history.

**Comment by SkafteNicki** [comment]
@Borda can you help fixing the commit history (you seem to be good at git :])?

**Comment by justusschock** [comment]
@ShomyLiu can you enable push access for maintainers? I tried to push to your branch and it was rejected

**Comment by ShomyLiu** [comment]
@justusschock  Hi, I have invited you into my branch, then you can push yourself. 
Thanks

**Comment by justusschock** [comment]
I rebased on master. So this should be fine now I think :)

**Comment by ShomyLiu** [comment]
@justusschock  Yeah, thanks for your effort. It is my first PR to lightning, I'm not so familiar with this whole merge process before.

**Comment by SkafteNicki** [comment]
Just waiting for test to pass, and then we can get this merged :]

**Review by SkafteNicki — APPROVED** [review]
👍

**Review by Borda — APPROVED** [review]
LGTM =)

**Pull request merged**
